--- ./src/libgm2/libcor/Makefile.am~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libcor/Makefile.am	2019-07-20 19:08:41.827125474 +0200
@@ -11,7 +11,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- ./src/libgm2/libcor/Makefile.in~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libcor/Makefile.in	2019-07-20 19:09:04.631281919 +0200
@@ -380,7 +380,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test $(MULTIOSDIR) != .; then echo /$(MULTIOSDIR); fi)
--- ./src/libgm2/libiso/Makefile.am~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libiso/Makefile.am	2019-07-20 19:09:20.055398547 +0200
@@ -11,7 +11,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- ./src/libgm2/libiso/Makefile.in~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libiso/Makefile.in	2019-07-20 19:09:48.035631457 +0200
@@ -396,7 +396,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test $(MULTIOSDIR) != .; then echo /$(MULTIOSDIR); fi)
--- ./src/libgm2/liblog/Makefile.am~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/liblog/Makefile.am	2019-07-20 19:09:57.287714323 +0200
@@ -11,7 +11,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- ./src/libgm2/liblog/Makefile.in~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/liblog/Makefile.in	2019-07-20 19:10:16.295893344 +0200
@@ -385,7 +385,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test $(MULTIOSDIR) != .; then echo /$(MULTIOSDIR); fi)
--- ./src/libgm2/libmin/Makefile.am~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libmin/Makefile.am	2019-07-20 19:10:42.940163506 +0200
@@ -11,7 +11,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- ./src/libgm2/libmin/Makefile.in~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libmin/Makefile.in	2019-07-20 19:10:53.552277128 +0200
@@ -383,7 +383,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test $(MULTIOSDIR) != .; then echo /$(MULTIOSDIR); fi)
--- ./src/libgm2/libpim/Makefile.am~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libpim/Makefile.am	2019-07-20 19:11:01.084359796 +0200
@@ -11,7 +11,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- ./src/libgm2/libpim/Makefile.in~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libpim/Makefile.in	2019-07-20 19:11:09.660455929 +0200
@@ -392,7 +392,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test $(MULTIOSDIR) != .; then echo /$(MULTIOSDIR); fi)
--- ./src/libgm2/libpth/Makefile.am~	2019-07-11 00:09:29.000000000 +0200
+++ ./src/libgm2/libpth/Makefile.am	2019-07-20 19:11:16.896538671 +0200
@@ -11,7 +11,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- ./src/libgm2/libpth/Makefile.in~	2019-07-11 00:09:29.000000000 +0200
+++ ./src/libgm2/libpth/Makefile.in	2019-07-20 19:12:01.565081356 +0200
@@ -304,7 +304,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test $(MULTIOSDIR) != .; then echo /$(MULTIOSDIR); fi)
--- ./src/libgm2/libulm/Makefile.am~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libulm/Makefile.am	2019-07-20 19:08:16.910977263 +0200
@@ -11,7 +11,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- ./src/libgm2/libulm/Makefile.in~	2019-07-08 17:12:57.000000000 +0200
+++ ./src/libgm2/libulm/Makefile.in	2019-07-20 19:08:03.650908456 +0200
@@ -395,7 +395,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(host_noncanonical)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test $(MULTIOSDIR) != .; then echo /$(MULTIOSDIR); fi)
