# FIXME: to investigate

--- a/src/configure.ac
+++ b/src/configure.ac
@@ -3638,7 +3638,7 @@ AC_SUBST(stage2_werror_flag)
 
 # Specify what files to not compare during bootstrap.
 
-compare_exclusions="gcc/cc*-checksum\$(objext) | gcc/ada/*tools/* | gcc/m2/gm2version\$(objext) | gcc/m2/*/M2Version\$(objext)"
+compare_exclusions="gcc/cc*-checksum\$(objext) | gcc/ada/*tools/* | gcc/m2/gm2version\$(objext) | gcc/m2/*/M2Version\$(objext) | gcc/SYSTEM\$(objext)"
 case "$target" in
   hppa*64*-*-hpux*) ;;
   hppa*-*-hpux*) compare_exclusions="gcc/cc*-checksum\$(objext) | */libgcc/lib2funcs* | gcc/ada/*tools/* | gcc/function-tests.o | gcc/m2/gm2version\$(objext) | gcc/m2/*/M2Version\$(objext)" ;;
