--- a/src/gcc/gm2/examples/gravity/testsim.py
+++ b/src/gcc/gm2/examples/gravity/testsim.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 #   Copyright (C) 2010
 #                 Free Software Foundation, Inc.
--- a/src/gcc/gm2/examples/gravity/testsim2.py
+++ b/src/gcc/gm2/examples/gravity/testsim2.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 #   Copyright (C) 2010
 #                 Free Software Foundation, Inc.
--- a/src/gcc/gm2/examples/pge/examples/boxes.py
+++ b/src/gcc/gm2/examples/pge/examples/boxes.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import pge, sys
 
--- a/src/gcc/gm2/examples/pge/examples/exampleBoxes.py
+++ b/src/gcc/gm2/examples/pge/examples/exampleBoxes.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import pge, sys
 # import pgemacro
--- a/src/gcc/gm2/examples/pge/examples/exampleBoxes.py~
+++ b/src/gcc/gm2/examples/pge/examples/exampleBoxes.py~
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import pge
 # import pgemacro
--- a/src/gcc/gm2/examples/pge/examples/triangle.py
+++ b/src/gcc/gm2/examples/pge/examples/triangle.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import pge, sys
 
--- a/src/gcc/gm2/examples/pge/fig.py
+++ b/src/gcc/gm2/examples/pge/fig.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import sys
 import getopt
--- a/src/gcc/gm2/examples/pge/max2code.py
+++ b/src/gcc/gm2/examples/pge/max2code.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import sys, getopt
 
--- a/src/gcc/gm2/examples/pge/pge.py
+++ b/src/gcc/gm2/examples/pge/pge.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import pgeif
 import pygame
--- a/src/gcc/gm2/examples/pge/pge2d.py
+++ b/src/gcc/gm2/examples/pge/pge2d.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 
 
--- a/src/gcc/gm2/examples/pge/pgeplayback.py
+++ b/src/gcc/gm2/examples/pge/pgeplayback.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import os, pygame, string, sys, getopt, math, struct, time
 from pygame.locals import *
--- a/src/gcc/gm2/examples/pge/pgepygame.py
+++ b/src/gcc/gm2/examples/pge/pgepygame.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import os, pygame, string, sys, getopt, math, struct
 from pygame.locals import *
--- a/src/gcc/gm2/examples/pge/t.py
+++ b/src/gcc/gm2/examples/pge/t.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import pygame
 from pygame.locals import *
--- a/src/gcc/gm2/examples/server/client.py
+++ b/src/gcc/gm2/examples/server/client.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import sys
 from socket import *
--- a/src/gcc/gm2/examples/swig/full-strlib/testequ.py
+++ b/src/gcc/gm2/examples/swig/full-strlib/testequ.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 #   Copyright (C) 2010
 #                 Free Software Foundation, Inc.
--- a/src/gcc/gm2/tools-src/array2index.py
+++ b/src/gcc/gm2/tools-src/array2index.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 # Copyright (C) 2010
 #               Free Software Foundation, Inc.
--- a/src/gcc/gm2/tools-src/def2texi.py
+++ b/src/gcc/gm2/tools-src/def2texi.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 # Copyright (C) 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008 2009,
 #               2010
--- a/src/gcc/gm2/tools-src/gensum.py
+++ b/src/gcc/gm2/tools-src/gensum.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # 
 # gensum.py a utility for summarizing the regression tests into html.
 # 
--- a/src/gcc/gm2/tools-src/tidydates.py
+++ b/src/gcc/gm2/tools-src/tidydates.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import os, sys
 
--- a/src/gcc/gm2/www/tools/texi2tr/src/navigation.py
+++ b/src/gcc/gm2/www/tools/texi2tr/src/navigation.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 # Copyright (C) 2011
 #               Free Software Foundation, Inc.
--- a/src/gcc/gm2/www/tools/texi2tr/src/outputdev.py
+++ b/src/gcc/gm2/www/tools/texi2tr/src/outputdev.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 # Copyright (C) 2011
 #               Free Software Foundation, Inc.
--- a/src/gcc/gm2/www/tools/texi2tr/src/texi2tr.py
+++ b/src/gcc/gm2/www/tools/texi2tr/src/texi2tr.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 # Copyright (C) 2011
 #               Free Software Foundation, Inc.
