--- a/src/gcc/gm2/Make-lang.in
+++ b/src/gcc/gm2/Make-lang.in
@@ -585,7 +585,7 @@ gm2m$(exeext): stage1/gm2/gm2m$(exeext)
 
 stage3/gm2/cc1gm2$(exeext): stage2/gm2/cc1gm2$(exeext) gm2/gm2-compiler-paranoid/m2flex.o \
                             $(P) $(GM2_C_OBJS) $(BACKEND) $(LIBDEPS) $(GM2_LIBS_PARANOID)
-	$(LINKER) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(GM2_C_OBJS) gm2/gm2-compiler-paranoid/m2flex.o \
+	$(LINKER) $(ALL_CFLAGS) $(LDFLAGS) -fno-lto -o $@ $(GM2_C_OBJS) gm2/gm2-compiler-paranoid/m2flex.o \
                             attribs.o \
                               $(GM2_LIBS_PARANOID) \
                               $(BACKEND) $(LIBS) \
@@ -593,7 +593,7 @@ stage3/gm2/cc1gm2$(exeext): stage2/gm2/c
 
 stage2/gm2/cc1gm2$(exeext): stage1/gm2/cc1gm2$(exeext) gm2/gm2-compiler/m2flex.o $(P) \
                             $(GM2_C_OBJS) $(BACKEND) $(LIBDEPS) $(GM2_LIBS)
-	$(LINKER) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(GM2_C_OBJS) gm2/gm2-compiler/m2flex.o \
+	$(LINKER) $(ALL_CFLAGS) $(LDFLAGS) -fno-lto -o $@ $(GM2_C_OBJS) gm2/gm2-compiler/m2flex.o \
                             attribs.o \
                               $(GM2_LIBS) \
                               $(BACKEND) $(LIBS) \
@@ -603,7 +603,7 @@ stage1/gm2/cc1gm2$(exeext): gm2/gm2-comp
                             $(P) $(GM2_C_OBJS) $(BACKEND) $(LIBDEPS) \
                             $(GM2_LIBS_BOOT) $(MC_LIBS)
 	$(__BREAKPOINT)
-	$(LINKER) $(ALL_CFLAGS) $(LDFLAGS) -o $@ $(GM2_C_OBJS) gm2/gm2-compiler-boot/m2flex.o \
+	$(LINKER) $(ALL_CFLAGS) $(LDFLAGS) -fno-lto -o $@ $(GM2_C_OBJS) gm2/gm2-compiler-boot/m2flex.o \
                             attribs.o \
                              $(GM2_LIBS_BOOT) $(MC_LIBS) \
                              $(BACKEND) $(LIBS) $(BACKENDLIBS)
