# DP: Ignore gm2 M2Version.o bootstrap comparison

--- a/src/configure.ac
+++ b/src/configure.ac
@@ -3573,6 +3573,7 @@
 # Specify what files to not compare during bootstrap.
 
 compare_exclusions="gcc/cc*-checksum\$(objext) | gcc/ada/*tools/*"
+compare_exclusions="$compare_exclusions | gcc/gm2/*/M2Version\$(objext) | gcc/gm2/gm2version\$(objext)"
 case "$target" in
   hppa*64*-*-hpux*) ;;
   hppa*-*-hpux*) compare_exclusions="gcc/cc*-checksum\$(objext) | */libgcc/lib2funcs* | gcc/ada/*tools/* | gcc/function-tests.o" ;;
