--- a/src/libgo/go/net/fd_posix.go	2020-08-03 15:12:53.000000000 +0200
+++ b/src/libgo/go/net/fd_posix.go	2020-09-24 16:03:50.000000000 +0200
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build aix darwin dragonfly freebsd linux netbsd openbsd solaris windows
+// +build aix darwin dragonfly freebsd hurd linux netbsd openbsd solaris windows
 
 package net
 
