diff -ur src/libgm2.orig/libm2cor/Makefile.am src/libgm2/libm2cor/Makefile.am
--- a/src/libgm2.orig/libm2cor/Makefile.am
+++ b/src/libgm2/libm2cor/Makefile.am
@@ -27,7 +27,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
diff -ur src/libgm2.orig/libm2cor/Makefile.in src/libgm2/libm2cor/Makefile.in
--- a/src/libgm2.orig/libm2cor/Makefile.in
+++ b/src/libgm2/libm2cor/Makefile.in
@@ -357,7 +357,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
diff -ur src/libgm2.orig/libm2iso/Makefile.am src/libgm2/libm2iso/Makefile.am
--- a/src/libgm2.orig/libm2iso/Makefile.am
+++ b/src/libgm2/libm2iso/Makefile.am
@@ -27,7 +27,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
diff -ur src/libgm2.orig/libm2iso/Makefile.in src/libgm2/libm2iso/Makefile.in
--- a/src/libgm2.orig/libm2iso/Makefile.in
+++ b/src/libgm2/libm2iso/Makefile.in
@@ -374,7 +374,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
diff -ur src/libgm2.orig/libm2log/Makefile.am src/libgm2/libm2log/Makefile.am
--- a/src/libgm2.orig/libm2log/Makefile.am
+++ b/src/libgm2/libm2log/Makefile.am
@@ -27,7 +27,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
diff -ur src/libgm2.orig/libm2log/Makefile.in src/libgm2/libm2log/Makefile.in
--- a/src/libgm2.orig/libm2log/Makefile.in
+++ b/src/libgm2/libm2log/Makefile.in
@@ -362,7 +362,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
diff -ur src/libgm2.orig/libm2min/Makefile.am src/libgm2/libm2min/Makefile.am
--- a/src/libgm2.orig/libm2min/Makefile.am
+++ b/src/libgm2/libm2min/Makefile.am
@@ -27,7 +27,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
diff -ur src/libgm2.orig/libm2min/Makefile.in src/libgm2/libm2min/Makefile.in
--- a/src/libgm2.orig/libm2min/Makefile.in
+++ b/src/libgm2/libm2min/Makefile.in
@@ -357,7 +357,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
diff -ur src/libgm2.orig/libm2pim/Makefile.am src/libgm2/libm2pim/Makefile.am
--- a/src/libgm2.orig/libm2pim/Makefile.am
+++ b/src/libgm2/libm2pim/Makefile.am
@@ -27,7 +27,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
diff -ur src/libgm2.orig/libm2pim/Makefile.in src/libgm2/libm2pim/Makefile.in
--- a/src/libgm2.orig/libm2pim/Makefile.in
+++ b/src/libgm2/libm2pim/Makefile.in
@@ -370,7 +370,7 @@
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(host_noncanonical)/$(version)
+libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
