# DP: recognize ppl version 1.0

--- a/src/configure.ac
+++ b/src/configure.ac
@@ -1577,8 +1579,10 @@
     CFLAGS="$CFLAGS $pplinc $gmpinc"
     AC_MSG_CHECKING([for version 0.11 (revision 0 or later) of PPL])
     AC_TRY_COMPILE([#include "ppl_c.h"],[
-    #if PPL_VERSION_MAJOR != 0 || PPL_VERSION_MINOR < 11
+    #if PPL_VERSION_MAJOR < 1
+    # if PPL_VERSION_MINOR < 11
     choke me
+    # endif
     #endif
     ], [AC_MSG_RESULT([yes])], [AC_MSG_RESULT([no]); ppllibs= ; pplinc= ; with_ppl=no ])
     CFLAGS="$saved_CFLAGS"
