# DP: Proposed patch for PR bootstrap/93409

--- a/src/libgomp/plugin/configfrag.ac
+++ a/src/libgomp/plugin/configfrag.ac
@@ -211,7 +211,7 @@
 	case "${target}" in
 	  x86_64-*-*)
 	    case " ${CC} ${CFLAGS} " in
-	      *" -m32 "*)
+	      *" -m32 "*|*" -mx32 "*)
 	        PLUGIN_HSA=0
 		;;
 	      *)
@@ -251,7 +251,7 @@
 	case "${target}" in
 	  x86_64-*-*)
 	    case " ${CC} ${CFLAGS} " in
-	      *" -m32 "*)
+	      *" -m32 "*|*" -mx32 "*)
 		PLUGIN_GCN=0
 		;;
 	      *)
--- a/src/libgomp/configure.ac
+++ b/src/libgomp/configure.ac
@@ -222,6 +222,7 @@
             [Define to 1 if building libgomp for an accelerator-only target.])
 fi
 
+# touch file for configfrag patch
 m4_include([plugin/configfrag.ac])
 
 # Check for functions needed.
