# DP: Revert the fix for PR tree-optimization/97043.

--- a/src/gcc/tree-vect-slp.c
+++ b/src/gcc/tree-vect-slp.c
@@ -2309,8 +2309,9 @@ vect_analyze_slp_instance (vec_info *vinfo,
 		  /* The load requires permutation when unrolling exposes
 		     a gap either because the group is larger than the SLP
 		     group-size or because there is a gap between the groups.  */
-		  && group_size == DR_GROUP_SIZE (first_stmt_info)
-		  && DR_GROUP_GAP (first_stmt_info) == 0)
+		  && (known_eq (unrolling_factor, 1U)
+		      || (group_size == DR_GROUP_SIZE (first_stmt_info)
+			  && DR_GROUP_GAP (first_stmt_info) == 0)))
 		{
 		  SLP_TREE_LOAD_PERMUTATION (load_node).release ();
 		  continue;
