# DP: Revert the fix for PR tree-optimization/97043.

Index: b/src/gcc/tree-vect-slp.c
===================================================================
--- a/src/gcc/tree-vect-slp.c
+++ b/src/gcc/tree-vect-slp.c
@@ -2309,8 +2309,9 @@ vect_analyze_slp_instance (vec_info *vin
 		  /* The load requires permutation when unrolling exposes
 		     a gap either because the group is larger than the SLP
 		     group-size or because there is a gap between the groups.  */
-		  && group_size == DR_GROUP_SIZE (first_stmt_info)
-		  && DR_GROUP_GAP (first_stmt_info) == 0)
+		  && (known_eq (unrolling_factor, 1U)
+		      || (group_size == DR_GROUP_SIZE (first_stmt_info)
+			  && DR_GROUP_GAP (first_stmt_info) == 0)))
 		{
 		  SLP_TREE_LOAD_PERMUTATION (load_node).release ();
 		  continue;
Index: b/src/gcc/ChangeLog
===================================================================
--- a/src/gcc/ChangeLog
+++ b/src/gcc/ChangeLog
@@ -227,13 +227,6 @@
 
 	* config/bpf/bpf.md ("nop"): Re-define as `ja 0'.
 
-2020-09-14  Richard Biener  <rguenther@suse.de>
-
-	PR tree-optimization/97043
-	* tree-vect-slp.c (vect_analyze_slp_instance): Do not
-	elide a load permutation if the current vectorization
-	factor is one.
-
 2020-09-14  Nathan Sidwell  <nathan@acm.org>
 
 	* config/i386/sse.md (mov<mode>): Fix operand indices.
