# DP: Use python3 explicitly

--- a/src/gcc/config/riscv/multilib-generator
+++ b/src/gcc/config/riscv/multilib-generator
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # RISC-V multilib list generator.
 # Copyright (C) 2011-2020 Free Software Foundation, Inc.
--- a/src/gcc/config/riscv/arch-canonicalize
+++ b/src/gcc/config/riscv/arch-canonicalize
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Tool for canonical RISC-V architecture string.
 # Copyright (C) 2011-2020 Free Software Foundation, Inc.
